pyomo/dae/tests/test_flatten.py

```diff
@@ -208,7 +292,997 @@ def c_rule(m, t):
+                        self._hashRef(Reference(m.b[...].v1)),
+                        }
+                self.assertEqual(len(ref_data), len(comps))
+                for comp in comps:
+                    self.assertIn(self._hashRef(comp), ref_data)
+            else:
+                raise RuntimeError()
+
+        # Space index specified:
+        indices = ComponentMap([(m.space, 4)])
+        sets_list, comps_list = flatten_components_along_sets(m, sets, Var,
+                indices=indices)
+
+        self.assertEqual(len(comps_list), len(sets_list))
+        self.assertEqual(len(sets_list), 1)
+
+        for sets, comps in zip(sets_list, comps_list):
+            if len(sets) == 2 and sets[0] is m.time and sets[1] is m.space:
+                # We descended into a block data that includes v2
+                self.assertEqual(len(comps), 3)
+
+                # Slices where we expect an attribute error somewhere,
+                # due to v2 being "skipped"
+                incomplete_slice = m.b[:,:].v2
+                incomplete_slice.attribute_errors_generate_exceptions = False
+                incomplete_ref = Reference(incomplete_slice)
+
+                ref_data = {
+                        self._hashRef(Reference(m.v[:,:])),
+                        self._hashRef(Reference(m.b[:,:].v1)),
+                        self._hashRef(incomplete_ref),
+                        }
+                self.assertEqual(len(ref_data), len(comps))
+                for comp in comps:
+                    self.assertIn(self._hashRef(comp), ref_data)
+            else:
+                raise RuntimeError()
+
+        # Time and space indices specified
+        indices = (3, 6)
+        sets_list, comps_list = flatten_components_along_sets(m, sets, Var,
+                indices=indices)
+
+        self.assertEqual(len(comps_list), len(sets_list))
+        self.assertEqual(len(sets_list), 1)
+
+        for sets, comps in zip(sets_list, comps_list):
+            if len(sets) == 2 and sets[0] is m.time and sets[1] is m.space:
+                # We descended into a block data that includes v2
+                self.assertEqual(len(comps), 3)
+
+                # Slices where we expect an attribute error somewhere,
+                # due to v2 being "skipped"
+                incomplete_slice = m.b[:,:].v2
+                incomplete_slice.attribute_errors_generate_exceptions = False
+                incomplete_ref = Reference(incomplete_slice)
+
+                ref_data = {
+                        self._hashRef(Reference(m.v[:,:])),
+                        self._hashRef(Reference(m.b[:,:].v1)),
+                        self._hashRef(incomplete_ref),
+                        }
+                self.assertEqual(len(ref_data), len(comps))
+                for comp in comps:
+                    self.assertIn(self._hashRef(comp), ref_data)
+            else:
+                raise RuntimeError()
+
+    def _model_4(self):
```

Comment:
Can you provide more descriptive model names (maybe incorporate some of the subsequent comment into the function name; possibly something like `_model4_slice_3_sets`)

Reply:
Done